src/plugins/data/public/query/filter_manager/filter_manager.test.ts

```diff
@@ -224,6 +231,40 @@ describe('filter_manager', () => {
       expect(newGlobalFilters).toHaveLength(2);
       expect(newAppFilters).toHaveLength(1);
     });
+
+    test('set filter with no state, and force pin', async () => {
+      const f1 = getFilter(esFilters.FilterStateStore.GLOBAL_STATE, false, false, 'age', 38);
+      f1.$state = undefined;
+
+      filterManager.setFilters([f1], true);
+      expect(filterManager.getGlobalFilters()).toHaveLength(1);
```

Comment:
small nit, and I'm don't really know what is happening here, so feel free to ignore, just thought maybe we could have a negative assetion here, too:

```ts
expect(filterManager.getGlobalFilters()).toHaveLength(1);
expect(filterManager.getAppFilters()).toHaveLength(0);
```

Reply:
Improved 👍